refactor(retrieval): configure bi-encoder attention mode (#1837)
NVIDIA-NeMo/Automodel: docs/model-coverage/embedding/mistralai/ministral3-bidirectional.mdx

Reviewer: ```suggestion
description: "Ministral3 embedding models in NeMo AutoModel, including the stock Hugging Face path, legacy bidirectional checkpoints, pooling, and recipes."
```

slug: model-coverage/embedding-models/mistralai/ministral3-bidirectional
---

NeMo AutoModel loads [Mistral AI's Ministral3](https://mistral.ai/news/ministraux/) with the stock Hugging Face `Ministral3Model` and configurable text attention for embedding and dense retrieval tasks. It defaults to `is_causal: false`, so each token can attend to both past and future tokens without requiring a custom model implementation.

> The encoder can be loaded directly from text-only checkpoints (e.g. `mistralai/Ministral-3B-Instruct`) and also automatically extracts the language model from Ministral3 VLM checkpoints (e.g. `mistralai/Ministral-3-3B-Base-2512` or `mistralai/Ministral-3-3B-Instruct-2512`). The custom `Ministral3BidirectionalModel` remains available through the legacy `ministral3_bidirec` model type for compatibility with existing checkpoints.

Reviewer: ```suggestion
The encoder can be loaded directly from text-only checkpoints (for example, `mistralai/Ministral-3B-Instruct`) and also automatically extracts the language model from Ministral3 VLM checkpoints (for example, `mistralai/Ministral-3-3B-Base-2512` or `mistralai/Ministral-3-3B-Instruct-2512`). The custom `Ministral3BidirectionalModel` remains available through the legacy `ministral3_bidirec` model type for compatibility with existing checkpoints.
```

> The encoder can be loaded directly from text-only checkpoints (e.g. `mistralai/Ministral-3B-Instruct`) and also automatically extracts the language model from Ministral3 VLM checkpoints (e.g. `mistralai/Ministral-3-3B-Base-2512` or `mistralai/Ministral-3-3B-Instruct-2512`). The custom `Ministral3BidirectionalModel` remains available through the legacy `ministral3_bidirec` model type for compatibility with existing checkpoints.

Reviewer: ```suggestion
The encoder can be loaded directly from text-only checkpoints (e.g., `mistralai/Ministral-3B-Instruct`) and also automatically extracts the language model from Ministral3 VLM checkpoints (e.g., `mistralai/Ministral-3-3B-Base-2512` or `mistralai/Ministral-3-3B-Instruct-2512`). The custom `Ministral3BidirectionalModel` remains available through the legacy `ministral3_bidirec` model type for compatibility with existing checkpoints.
```

| | |
|---|---|
| **Tasks** | Embedding, Dense Retrieval |
| **Architecture** | `Ministral3Model` with configurable attention (preferred); `Ministral3BidirectionalModel` (legacy) |
| **Parameters** | 3B |
| **HF Org** | [mistralai](https://huggingface.co/mistralai) |

## Available Models

Any Ministral3 checkpoint can be loaded with bidirectional or causal attention. The following configurations are tested:

> - **Ministral-3-3B-Base-2512** — VLM checkpoint, language model is extracted automatically
> - **Ministral-3-3B-Instruct-2512** — VLM checkpoint, language model is extracted automatically

Reviewer: ```suggestion
- **Ministral-3-3B-Base-2512**: VLM checkpoint, language model is extracted automatically
- **Ministral-3-3B-Instruct-2512**: VLM checkpoint, language model is extracted automatically
```
